S4MK3: stop wheel led blinking when track is over/stopped

acolombier opened the pull request:
Fixes #13995

Arguably, the code could be slightly cleaner, but I wanted to submit a fix for this before the 2.5 release.

@ronso0 could you please have a look?

acolombier committed:
S4MK3: stop wheel led blinking when track is over/stopped

ronso0 commented on res/controllers/Traktor-Kontrol-S4-MK3.js:
```diff
@@ -2523,8 +2523,25 @@ class S4Mk3Deck extends Deck {
+                    const connection1 = engine.makeConnection(this.group, "play", (play) => this.output.bind(this)(engine.getValue(this.group, "playposition"), play));
+                    // This is useful for case where effect would have been fully disabled in Mixxx. This appears to be the case during unit tests.
+                    if (connection1) {
+                        this.outConnections[0] = connection1;
```
Do I understand this correctly:
this will replace `connection0`?

acolombier replied:
Ah that's a bug :sweat_smile:

ronso0 commented:
There's now a small glitch:
when I stop the deck in eot range, the LED spot will briefly light up at the previous position.
* play deck
* seek into eot range
* stop deck (note LED spot position = pos1)
* play (flash)
* stop: spot briefly at pos1, then at correct new position

acolombier commented:
I will look into this when I can

ronso0 commented:
Sure.
It's working as desired btw : )

acolombier commented:
> There's now a small glitch

Looks like this is a hardware limitation. I've debug the HID messages, and what we are saying is good. 
It looks like there is a built-in animation management of the controller, which is managed individual for each wheel lighting mode. It seems this this animation is enlightening the delta between two pos in order create a transition animation. 

I have added a nasty workaround that basically makes the backlit off during the animation and then restore the colour. This mean that LED stays off for about ~200ms, which seems to be the animation length. Or we could keep the limitation as is?

I have also added another commit which display the LED spot again if the wheel is touched inside the EOT range, and also keep the wheel off is no track is loaded.

ronso0 commented:
Thank you looking into this glitch and finishing this!

Works nicely now!

ronso0 commented:
Squash please, then this is ready to roll!

acolombier commented:
Done! Thanks for looking into this @ronso0
